iron-fit-behavior.html

```diff
@@ -368,33 +370,30 @@
         }
       }
 
-      left += this._fitInfo.margin.left;
-      top += this._fitInfo.margin.top;
+      left += margin.left;
+      top += margin.top;
 
       // Use original size (without margin)
-      var right = left + rect.width;
-      var bottom = top + rect.height;
-
-      left = Math.max(left, this._fitInfo.margin.left);
-      top = Math.max(top, this._fitInfo.margin.top);
-
-      var maxWidth = Math.min(fitRect.right - this._fitInfo.margin.right, right) - left;
-      var maxHeight = Math.min(fitRect.bottom - this._fitInfo.margin.bottom, bottom) - top;
+      var right = Math.min(fitRect.right - margin.right, left + rect.width);
+      var bottom = Math.min(fitRect.bottom - margin.bottom, top + rect.height);
 
       var minWidth = this._fitInfo.sizedBy.minWidth;
       var minHeight = this._fitInfo.sizedBy.minHeight;
-
-      if (maxWidth < minWidth || right < minWidth) {
-        left = -9999;
-        maxWidth = 0;
+      if (left < margin.left) {
+        left = margin.left;
+        if (right - left < minWidth) {
+          left = right - minWidth;
+        }
       }
-      if (maxHeight < minHeight || bottom < minHeight) {
-        top = -9999;
-        maxHeight = 0;
+      if (top < margin.top) {
+        top = margin.top;
+        if (bottom - top < minHeight) {
+          top = bottom - minHeight;
+        }
```

Comment:
yes we did..! 
Also, relying on setting only `left, top` turned out to be a good thing because it doesn't have any problem with scaling factors (e.g. issues like this one https://github.com/PolymerElements/iron-dropdown/issues/76)